### DISCUSSION TITLE
do we want to nickname features?

### DISCUSSION THREAD

**Discussion by myjaylee1123** [arch-kiosk/arch-kiosk-office#1213] [General]
sometimes it's hard to go back and find a specific feature you were working on because they're all identified only by date created or by the ID number (which, now that there are so many features, there's no real logic to how they're generated on a UI level). this can be a little annoying when you need to reference a specific feature but you forgot the name of it and need to go hunting for it. 

sam suggested the formate of ID#xyz "nickname" when referring to a feature. this way, we always associate the nickname with an ID number so that we still refer to a feature by ID name but not by just nickname.

**Comment by SamKimball-BrownU** [comment]
in order to get users to use the id along with nickname (instead of just nickname), perhaps there should be some type of menu? where you search up the nickname, and are given the id in return. Perhaps in the banner, next to the pinboard.... 

This is, of course, if we do think that nicknames are a good idea. Some projects certainly use them, where others are against it. It should be configurable, whether to have the option to input nicknames or not, I think.

**Comment by SamKimball-BrownU** [comment]
If we do implement nicknames, we would also want to implement it for other units. Survey units, excavation units, etc. 

Distinction between alternative numbers (which is used, if I'm correct, to accommodate mostly legacy data) vs nicknames

**Comment by osiegel14** [comment]
I agree that nicknames would be a useful option to have. This is perhaps a silly comment, but if we eventually intend to allow users to search for specific tags, tags themselves could be a way to allow users to input nicknames for trenches, survey areas, features, or even contexts without needing to add or program a specific nickname feature. On the other hand, it seems like one of the issues that came up in relation to tags and features last week is that we (or at least I) am still unclear on what exactly we want tagging to accomplish, and what falls beyond its purview. I'm wary of offloading even more responsibility onto an element that we haven't fully defined yet.

**Comment by urapadmin** [comment]
tags exist to group loci. Even if we expand them, their purpose is to group several archaeological entities. A nickname addresses exactly one. So tags are not an appropriate tool.
If we really want nicknames - and I am less confident that it is so useful - we need to define where we want them first. Should a unit have a nickname, or a locus? A collected material, even a sherd?

**Comment by luizaogs** [comment]
I am also a bit skeptical of the usefulness of this, and can even see it being problematic if nicknames end up being used more often than id numbers (even if they appear together in the recording system itself). But it seems like people did miss it during survey, and in that case since one doesn't have anything like a dayplan with numbers that can be pinned to the pinboard and referred to it seems more applicable (though also, I wonder if that is because we had not really changed the list of available feature types to fit our needs, and so we each probably had a million "others" that should have been "benches." But having a million "benches" when trying to find a specific bench would also be a bit difficult to navigate in that list, I suppose). 

Anyway -- I was then going to say that having nicknames in anything other than units seems like overkill (even for loci -- you have all of the information you might need to distinguish them at your fingertips in the little menu right below the loci list). But then I ventured into the wiki to see if this had been discussed previously, and it looks like even at Uronarti nicknames for collected materials were deemed potentially necessary? Hm. In that case I would probably suggest making the use of nicknames for different recording categories configurable by project. Indeed as Sam suggests above, but not even just the ability to turn nicknames on/off, but rather making it so that if a specific project needs to have nicknames for collected materials they can turn that on. Otherwise I worry a bit that every sherd will have a nickname and then things might get overwhelming (and/or ridiculous).
